modules/visual_script/editor/visual_script_property_selector.cpp

```diff
@@ -42,6 +42,15 @@
 #include "scene/main/node.h"
 #include "scene/main/window.h"
 
+// This helper returns a String containing the given String, with "res://" removed from the beginning, if present.
+static String strip_resource_schema(const String &from) {
+	if (from.begins_with("res://")) {
+		return from.substr(6, from.length());
+	} else {
+		return from;
+	}
```

Comment:
I saw you were asking on RC about adding comments. So for me comments like this one are just a noise. Method is trivial and I think it's obvious what it does just after a brief look at the code. I'd say don't tell what the code does unless: it's not easy to understand, or you're referring to some bigger chunk of code in a summary-like manner, or you're sectioning code with a title-like comments, etc. Simple comments to help with quick glancing at the code and more detailed ones when explaining how something works / why it's done like that (but again: no reason to explain how 2+2 works as well as no reason to tell it's an addition). Of course there's no golden rule and that's just my opinion. :wink:

Also there's `String::trim_prefix` you can use:

```suggestion
static String strip_resource_schema(const String &from) {
	return from.trim_prefix("res://");
```

I don't know if it's desired to have such one-liner specialization of `trim_prefix`. No idea whether there's some preference / guidelines about it.

Reply:
Thanks, I looked for ```trim_prefix``` and did not find it.   Note that throughout the code base, people are incorrectly using ```replace_first``` instead.   I don't think it makes any sense to have the specialization since the code is gone.  I will just go back to the original code and just use the correct function instead of ```ltrim```.  This was just something I fixed up real quick because the code doesn't run so I could not work on a bug I am investigating.

The comment here is trivial, I agree.  I may have left that in just to see what would happen if I submit something with a super simple comment.